proposals/testing/NNNN-image-attachments-in-swift-testing-windows.md

```diff
@@ -0,0 +1,381 @@
+  /// `nil` to let the testing library decide which image format to use. If you
+  /// pass `nil`, then the image format that the testing library uses depends on
+  /// the path extension you specify in `preferredName`, if any. If you do not
+  /// specify a path extension, or if the path extension you specify doesn't
+  /// correspond to an image format the operating system knows how to write, the
+  /// testing library selects an appropriate image format for you.
+  public static func record<T>(
+    _ image: T,
+    named preferredName: String? = nil,
+    as imageFormat: AttachableImageFormat? = nil,
+    sourceLocation: SourceLocation = #_sourceLocation
+  ) where T: AttachableAsIWICBitmapSource, AttachableValue == _AttachableImageWrapper<T>
+}
+```
+
+> [!NOTE]
+> `_AttachableImageWrapper` was described in [ST-0014](https://github.com/swiftlang/swift-evolution/blob/main/proposals/testing/0014-image-attachments-in-swift-testing-apple-platforms.md#attaching-a-conforming-image).
+> The only difference on Windows is that its associated `Image` type is
+> constrained to `AttachableAsIWICBitmapSource` instead of `AttachableAsCGImage`.
+
+### Specifying image formats
+
+As on Apple platforms, a test author can specify the image format to use with
+`AttachableImageFormat`. See [ST-0014](https://github.com/swiftlang/swift-evolution/blob/main/proposals/testing/0014-image-attachments-in-swift-testing-apple-platforms.md#specifying-image-formats)
+for more information about that type.
+
+Windows does not use Uniform Type Identifiers, so those `AttachableImageFormat`
+members that use `UTType` are not available here. Instead, Windows uses a
+variety of COM classes that implement codecs for different image formats.
+Conveniences over those COM classes' `CLSID` values are provided:
+
+```swift
+extension AttachableImageFormat {
+  /// The `CLSID` value corresponding to the WIC image encoder for this image
+  /// format.
+  public var clsid: CLSID { get }
+
+  /// Construct an instance of this type with the given `CLSID` value and
+  /// encoding quality.
+  ///
+  /// - Parameters:
+  ///   - clsid: The `CLSID` value corresponding to a WIC image encoder to use
+  ///     when encoding images.
+  ///   - encodingQuality: The encoding quality to use when encoding images. For
+  ///     the lowest supported quality, pass `0.0`. For the highest supported
+  ///     quality, pass `1.0`.
+  ///
+  /// If the target image encoder does not support variable-quality encoding,
+  /// the value of the `encodingQuality` argument is ignored.
+  ///
+  /// If `clsid` does not represent an image encoder type supported by WIC, the
+  /// result is undefined. For a list of image encoders supported by WIC, see
+  /// the documentation for the [`IWICBitmapEncoder`](https://learn.microsoft.com/en-us/windows/win32/api/wincodec/nn-wincodec-iwicbitmapencoder)
+  /// class.
+  public init(_ clsid: CLSID, encodingQuality: Float = 1.0)
+}
```

Comment:
I should also add support for creating an instance of this type from a container format GUID:

```swift
public init(containerFormatGUID guid: GUID, encodingQuality: Float = 1.0)
```

See https://github.com/swiftlang/swift-testing/issues/1294